eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -49,25 +56,33 @@ jobs:
         OSVmImage: 'ubuntu-18.04'
         VcpkgInstall: 'curl[ssl] libxml2 openssl'
         VCPKG_DEFAULT_TRIPLET: 'x64-linux'
-        CmakeArgs: ' -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON'
+        CmakeArgs: ' -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DCMAKE_BUILD_TYPE=Debug -DBUILD_CODE_COVERAGE=ON'
+        AptDependencies: 'gcovr lcov'
+        CODE_COVERAGE: '${{ parameters.Coverage }}'
```

Comment:
Do you really only want to enable code coverage on some pipelines or do you want to do it only for one matrix entry for all the pipelines? I'd assume you want the later which is what most of the languages do. In which case you don't need to set this to the template parameter but instead just pass in true/enable here always.

Reply:
Ideally, it would be for one matrix entry.
However, I need to support disabling it for some pipelines (storage and identity services) becasuse they are either not running any tests or it's broken.
That's why I made this opt it from the top.

The other approch is to make code coverage steps generation to handle when there are no tests/targets to invoke to generate data, and storage and identity would do a no-op on those steps. 

But I prefered the first one where we explicitly know we are disabling it.  The second option can lead to a bug where step is not working and we don't realized soon because it does not fail

Reply:
I'd like us to be like other languages unless there is a reason not to.  Storage/Identity need to be fixed ASAP.  @vhvb1989 please work with Wes to get us in alignment with the other languages.

Reply:
I think our goal should be to run it for every pipeline and if we have some that don't work today I'd prefer that do an opt-out, with a tracking issue, instead of opting in for the ones that currently do support it.

Reply:
@weshaggard can you suggest how to do the opt out?

These are 3 steps from matrix-job. We would want to disable just one or two.

The way I think about it is with an opt in the ci to set `disable_code_coverage` that we can use to disable those steps only for that pipeline. (Basically the same but inverted `turn-off` vs `turn-on`).  Would you prefer it like that?

Or would you have something else as suggestion

Reply:
Yes I would prefer a disable option in the pipelines that currently don't support it. Something like:

```
// Set enabled by default. 
Coverage: 'enabled' 
...

// add condition around this entry which will complete remove it for those places that opt-out.
   - ${{ if eq(parameters.Coverage, 'enabled') }}
      Linux_x64_with_unit_test:
        OSVmImage: 'ubuntu-18.04'
        VcpkgInstall: 'curl[ssl] libxml2 openssl'
        VCPKG_DEFAULT_TRIPLET: 'x64-linux'
        CmakeArgs: ' -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON'
        CmakeArgs: ' -DBUILD_TESTING=ON -DRUN_LONG_UNIT_TESTS=ON -DCMAKE_BUILD_TYPE=Debug -DBUILD_CODE_COVERAGE=ON'
        AptDependencies: 'gcovr lcov'
        CODE_COVERAGE: enabled


// In the ci.yml files that don't support coverage yet pass in:

Coverage: 'disabled'
```

Reply:
done